utils/mountd/cache.c

```diff
@@ -1429,6 +1435,52 @@ int cache_process_req(fd_set *readfds)
 	return cnt;
 }
 
+/*
+ * Expire export cache entries in path which are subdirectories of
+ * exp->m_export.e_path. For instance, if exp->m_export.e_path is
+ *     /a/b
+ * and path is
+ *     /a/b/c/d/e
+ * this will expire the export cach entries for
+ *     /a/b/c
+ *     /a/b/c/d
+ *     /a/b/c/d/e
+ */
+int cache_expire_subdirs(nfs_export *exp, char *path)
+{
+	char buf[RPC_CHAN_BUF_SIZE];
+
+	char *exp_path = exp->m_export.e_path;
+	char *domain = exp->m_client->m_hostname;
+
+	/* Nothing to do if path isn't a subdirectory of exp_path. */
+	unsigned int l = strlen(exp_path);
+	if (strlen(path) <= l || path[l] != '/' ||
+	    strncmp(exp_path, path, l) != 0)
+		return 0;
+
+	int f = open("/proc/net/rpc/nfsd.export/channel", O_WRONLY);
+	if (f < 0)
+		return -1;
+	while (path[l]) {
+		l++;
+		while (path[l] != '/' && path[l])
+			l++;
+		char saved = path[l];
+		path[l] = 0;
+		xlog(L_WARNING, "expiring %s\n", path);
+		/*
+		 * A negative TTL will cause the entry to be immediately
+		 * expired.
+		 */
+		dump_to_cache(f, buf, sizeof(buf), domain, path, NULL, -10);
```

Comment:
Is the -10 TTL value significant here or will any negative value work? If the -10 value is significant then we should document why.

Reply:
There isn't any particular significance to -10. I think anything negative should work, but I picked something more negative than -1 just in case there are edge cases in the expiration/flushing logic in the cache.

Reply:
Ok, maybe drop a comment here so that we indicate that we want a timer that is sufficiently in the past.

Reply:
👍 done